rcljava/src/main/cpp/org_ros2_rcljava_action_ActionServerImpl_GoalHandleImpl.cpp

```diff
@@ -81,10 +82,39 @@ JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_00024GoalHandleImpl_native
   return static_cast<jint>(status);
 }
 
-static void update_goal_state(JNIEnv * env, jlong jgoal_handle, rcl_action_goal_event_t event)
+JNIEXPORT void
+JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_00024GoalHandleImpl_nativeUpdateGoalState(
+  JNIEnv * env, jclass, jlong jgoal_handle, jlong jto_status)
 {
-  rcl_action_goal_handle_t * goal_handle = reinterpret_cast<rcl_action_goal_handle_t *>(
-    jgoal_handle);
+  rcl_action_goal_event_t event = GOAL_EVENT_NUM_EVENTS;
+  switch (jto_status) {
+    case GOAL_STATE_EXECUTING:
+      event = GOAL_EVENT_EXECUTE;
+      break;
+    case GOAL_STATE_CANCELING:
+      event = GOAL_EVENT_CANCEL_GOAL;
+      break;
+    case GOAL_STATE_SUCCEEDED:
+      event = GOAL_EVENT_SUCCEED;
+      break;
+    case GOAL_STATE_CANCELED:
+      event = GOAL_EVENT_CANCELED;
+      break;
+    case GOAL_STATE_ABORTED:
+      event = GOAL_EVENT_ABORT;
+      break;
+    case GOAL_STATE_ACCEPTED:  // fallthrough
+    case GOAL_STATE_UNKNOWN:  // fallthrough
+    default:
+      break;
+  }
+  if (event >= GOAL_EVENT_ABORT) {
```

Comment:
This looks incorrect. I think `GOAL_EVENT_ABORT` is valid. Maybe this should be 
```suggestion
  if (event >= GOAL_EVENT_NUM_EVENTS) {
```

Reply:
Or, could we just throw a Java exception and return in the `default` case above?

Reply:
Good catch, fixed in https://github.com/osrf/ros2_java/pull/65/commits/0ce939cc25e1053b0989b26694ff4b738caea54e.